src/Analysis/Ast/Impl/Specializations/BuiltinsSpecializations.cs

```diff
@@ -135,8 +135,11 @@ public static IMember List(IPythonInterpreter interpreter, IPythonFunctionOverlo
                 def = args[2];
             }
 
+            // second argument to getattr was not a string, which is a runtime error
+            // getattr(a, 3.14)
             if (name == null) {
-                return def;
+                // TODO diagnostic error when second arg of getattr is not a string
+                return  module.Interpreter.UnknownType;
```

Comment:
Nit: formatting (extra space)

I'm not sure I like this change; I know that if we believe the name argument wasn't a string then it's wrong, but returning unknown as a whole feels worse than inferring at least _something_ via the default.

Reply:
So the reason I made it Unknown was because a test checked for it to be UnknownType when I was returning NoneType. Would it preferable to return NoneType? 

Doing ```getattr()``` is a runtime error and our convention has been to return UnknownTypes for those.

Reply:
What test is failing? Presumably, all tests passed before this PR, so the code I had written would already have been passing.

Reply:
The test actually was already saying that it should be unknown (ClassesTest.GetAttr). Resolving.